Helper/Entity/Product/PriceManager/ProductWithChildren.php

```diff
@@ -158,28 +158,50 @@ protected function setFinalGroupPrices($field, $currencyCode, $min, $max, $dashe
     {
         if (count($subproducts) > 0) {
             $groupPriceList = [];
+            $subProductsMin = -1;
+            $subProductsMax = 0;
             /** @var Group $group */
             foreach ($this->groups as $group) {
                 $groupId = (int) $group->getData('customer_group_id');
                 $minPrice = $min;
+
                 foreach ($subproducts as $subProduct) {
                     $subProduct->setData('customer_group_id', $groupId);
                     $subProduct->setData('website_id', $subProduct->getStore()->getWebsiteId());
                     $specialPrice = $this->getSpecialPrice($subProduct, $currencyCode, $withTax, []);
                     $tierPrice = $this->getTierPrice($subProduct, $currencyCode, $withTax);
                     $price     = $this->getTaxPrice($product, $subProduct->getPriceModel()->getFinalPrice(1, $subProduct), $withTax);
+
                     if (!empty($tierPrice[$groupId]) && $specialPrice[$groupId] > $tierPrice[$groupId]){
                         $minPrice = $tierPrice[$groupId];
                     }
-                    $groupPriceList[$groupId]['min'] = min($minPrice, $price);
-                    $groupPriceList[$groupId]['max'] = max($max, $price);
+
+                    if (isset($tierPrice[$groupId]) && $tierPrice[$groupId] !== false) {
```

Comment:
Does this conflict with the logic that immediately precedes it? Perhaps the conditionals could be merged?

Reply:
TBH I think this is not even needed, I guess it was one of the many tests I made and I forgot to remove it, sorry for that.